library/core/src/slice/index.rs

```diff
@@ -526,3 +528,89 @@ unsafe impl<T> SliceIndex<[T]> for ops::RangeToInclusive<usize> {
         (0..=self.end).index_mut(slice)
     }
 }
+
+/// Convert pair of `ops::Bound`s into `Range` without performing any bounds checking and (in debug) overflow checking
+fn into_range_unchecked(
+    len: usize,
+    (start, end): (ops::Bound<usize>, ops::Bound<usize>),
+) -> ops::Range<usize> {
+    let start = match start {
+        Bound::Included(i) => i,
+        Bound::Excluded(i) => i + 1,
+        Bound::Unbounded => 0,
+    };
+    let end = match end {
+        Bound::Included(i) => i + 1,
+        Bound::Excluded(i) => i,
+        Bound::Unbounded => len,
+    };
+    Range { start, end }
+}
+
+/// Like `check_range` except it returns `None` on invalid ranges.
+fn into_maybe_range(
```

Comment:
I think we could call this `checked_into_range`, and the function above simply `into_range`.

Reply:
Maybe, but the name `into_range_unchecked` was used since in can be easily used to screw things up.

Reply:
Sorry this is kinda off-topic, but I've seen code like in `into_range_unchecked` in a few projects including my own. It would be nice if there were exposed as a public `Range::from_bounds(impl RangeBounds) -> Self` constructor.

Reply:
The only thing `into_range_unchecked` isn't doing is checked arithmetic on the bounds (assuming `into_maybe_range` doesn't actually need to do bounds checking itself). The source of unsafety is in how we use it in `get_unchecked` and `get_unchecked_mut`, where the caller becomes responsible for ensuring they're valid.

I would be ok with `into_range_unchecked` and `into_range` though.

Reply:
Done with the names you suggested

Reply:
@Kimundi That exists now:  
https://doc.rust-lang.org/std/ops/trait.RangeBounds.html#method.assert_len

However, if you'll be using it, you should know that its name might change: #81154